### ISSUE TITLE
Safety Evaluation when Crossing Intersections for People with Disabilities

### ISSUE THREAD

**Issue by RayZ5566** [lily71903/stanCode_AI_Project#1]
### Introductions:

For ordinary people crossing intersections is inevitable in daily live. However, people with disabilities (eg. visually or physically impaired) might have difficulty to make the appropriate decision with if the intersection is safe to cross. With the development of artificial intelligence and CNNs, the detection of traffic light is no longer a unsolvable problem and even the detection model can already make a accurate prediction(Muller et al 2018 ; Gupta et al, 2019). Nowadays, this kind of technology, detecting and extracting features from the real time road condition, is primarily applied to the evolving of autonomous vehicles. From the further investigating in this technique, we barely found the implementation in assisting the visually impaired people purpose.
In this project, we propose a vision-based model for detecting the real-time intersection condition. The model will detect the traffic light, the current circumstance of the intersection and if the zebra crossing is clear to walk, especially for blinds. By analyzing with Deep Learning model from images of intersections, the model could evaluate the current overall conditions and provide prompt objective overall feedback, the outcome will be either pass or stop.

### Problem Statements:

We will investigate various conditions of intersections would meet and utilizing the CNN model to evaluate if the intersection is safe to pass or not (the width of the road is considered). The model will be trained with real life intersections photos mainly located in Taipei (the shape of the traffic light for pedestrians is not round and distinct from ordinary, not stable). 
The all training data include two factors, the traffic light and zebra crossing.
The expected results will be two parts, not safe to walk(0) and safe to walk(1), and the width of the crossing road(crossing time) showing the predicted probability of traffic light, green or red.

### Approach:

1. Model: provided CNN pre-trained model => comparing four main pretrained models, VGG19, Inceptionv3, ResNet50, EfficientNetV2 and the best preformed one will be selected.
2. Detecting the traffic light in the image and predicting the probability of green or red
3. Detecting the circumstance of on the zebra crossing if it is clear (only vehicles and scooters will be consider as obstacles).
4. Evaluating the time under walking pace required to cross the intersection => it will  be evaluated by the width of the road

### Preliminary Results:

1. Dataset: 
    1. Training data: Total: 1070, not safe to walk(0):535, safe to walk(1): 535. Figure 1. shows the distribution of all data and Figure 2. shows the basic statistical information of training and validation data.
    2. Data collection: All real life intersection photos were taken by 4 group members located in 4 distinct area(3 Taipei 1 Taoyuan).
2. Data Preprocessing:
    1. Attaching labels on each intersection photo
        1. labels: not safe to walk(0), safe to walk(1); width of the road(meter, float round to 1)
    2. Converting image file format, all format to jpg
    3. Cropping photos into square shape
    4. Resize cropped photos => downsize the data, reduce the capacity required and boost the model speed

<img src="https://user-images.githubusercontent.com/62705571/121795179-ecef2180-cc40-11eb-9fd3-fcf7cd328de6.png" width="600" height="400">
Figure 1. The distribution of the dataset
<img src="https://user-images.githubusercontent.com/62705571/121795183-fc6e6a80-cc40-11eb-8907-ec7fe39652b0.png" width="400" height="800">
Figure 2. the basic statistical information of training and validation data

 

### Reference:

1. A. Gupta and A. Choudhary, "A Framework for Traffic Light Detection and Recognition using Deep Learning and Grassmann Manifolds," 2019 IEEE Intelligent Vehicles Symposium (IV), 2019, pp. 600-605, doi: 10.1109/IVS.2019.8814062. https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=8814062
2.  Julian Muller  and Klaus Dietmayer. Detecting Traffic Lights by Single Shot Detection https://arxiv.org/pdf/1805.02523.pdf ,  arXiv:1805.0252, 2018
